components/core/src/clp_s/ArchiveWriter.cpp

```
@@ -65,12 +94,130 @@
+) {
+    FileReader reader;
+    for (auto const& file : files) {
+        std::string file_path = m_archive_path + file.n;
+        reader.open(file_path);
+        char read_buffer[cReadBlockSize];
+        while (true) {
+            size_t num_bytes_read{0};
+            ErrorCode const error_code
+                    = reader.try_read(read_buffer, cReadBlockSize, num_bytes_read);
+            if (ErrorCodeEndOfFile == error_code) {
+                break;
```

cppcheck (2.10-2)

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)